src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -36,6 +39,30 @@ public static Index parseIndex(String oneBasedIndex) throws ParseException {
         return Index.fromOneBased(Integer.parseInt(trimmedIndex));
     }
 
+    /**
+     * Parses a string of {@code oneBasedIndex}es seperated by spaces into a list of {@code Index} of exactly size
+     * {@code parseCount} and returns it. Leading and trailing whitespaces will be trimmed.
+     *
+     * @param parseCount The exact amount of indexes to parse.
+     * @throws ParseException if the specified indexes is invalid (not non-zero unsigned integer), or if the amount of
+     *     indexes is not exactly {@code parseCount}
+     */
+    public static List<Index> parseIndexes(String oneBasedIndexes, int parseCount) throws ParseException {
+        String indexSeparator = " ";
+        String trimmedIndexes = oneBasedIndexes.trim();
+        String[] indexes = trimmedIndexes.split(indexSeparator);
+
+        List<Index> indexList = new ArrayList<>();
+        for (String index : indexes) {
+            indexList.add(parseIndex(index));
+        }
+
+        if (indexList.size() != parseCount) {
+            throw new ParseException(MESSAGE_UNEXPECTED_INDEX_COUNT);
+        }
```

Comment:
Would be better to move it upwards before parsing the indexes to make the happy path prominent.

Reply:
That's a nice suggestion